src/Mcfedr/AwsPushBundle/Service/Devices.php

```diff
@@ -34,25 +34,31 @@ public function __construct(SnsClient $client, $platformARNS)
      *
      * @param string $deviceId device token
      * @param string $platform platform on which to register
+     * @param string $userData
      * @return string the endpoint ARN for this device
      * @throws PlatformNotConfiguredException
+     * @throws \Exception
      */
-    public function registerDevice($deviceId, $platform)
+    public function registerDevice($deviceId, $platform, $userData = null)
     {
         if (!isset($this->arns[$platform])) {
             throw new PlatformNotConfiguredException("There is no configured ARN for $platform");
         }
 
         try {
-            $res = $this->sns->createPlatformEndpoint(
-                [
-                    'PlatformApplicationArn' => $this->arns[$platform],
-                    'Token' => $deviceId,
-                    'Attributes' => [
-                        'Enabled' => 'true'
-                    ]
+            $args = [
+                'PlatformApplicationArn' => $this->arns[$platform],
+                'Token' => $deviceId,
+                'Attributes' => [
+                    'Enabled' => 'true'
                 ]
-            );
+            ];
+
+            if ($userData && is_string($userData)) {
```

Comment:
So should $deviceId and $platform
its in the doc, i dont think we should be checking it